src/components/notifications/notification.resolver.ts

```diff
@@ -0,0 +1,32 @@
+import { Args, Mutation, Query, Resolver } from '@nestjs/graphql';
+import {
+  ListArg,
+  LoggedInSession,
+  NotImplementedException,
+  Session,
+} from '~/common';
+import {
+  MarkNotificationReadArgs,
+  Notification,
+  NotificationList,
+  NotificationListInput,
+} from './dto';
+
+@Resolver()
+export class NotificationResolver {
+  @Query(() => NotificationList)
+  async notifications(
+    @LoggedInSession() session: Session,
+    @ListArg(NotificationListInput) input: NotificationListInput,
+  ): Promise<NotificationList> {
+    throw new NotImplementedException().with(input, session);
+  }
+
+  @Mutation(() => Notification)
+  async markNotificationRead(
```

Comment:
Actually I think 
```suggestion
  async readNotification(
```
would be better.

Reply:
Eh, the problem with `readNotification` is it's ambiguous.   Is "read" a past tense verb, a present tense verb, or a noun here without context?

Reply:
True. But this is in the context of a mutation action?

Reply:
Point.  I can get on board with it.